web/modules/custom/dpl_login/src/Adgangsplatformen/Config.php

```diff
@@ -108,4 +88,21 @@ public function getClientSecret(): string {
     return $this->getValue('client_secret');
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  public function getConfig(): array {
+    $settings = $this->configManager->getConfigFactory()->get(self::CONFIG_KEY)->get('settings');
```

Comment:
What if `$this->configManager->getConfigFactory()->get(self::CONFIG_KEY)` returns NULL? Wont `->get` fail?

Maybe you could get around it with just:

```
$settings = $this->configManager->getConfigFactory()->get(self::CONFIG_KEY)?->get('settings');

Reply:
> What if $this->configManager->getConfigFactory()->get(self::CONFIG_KEY) returns NULL

[It does not. It will always return an `\Drupal\Core\Config\ImmutableConfig` object.](https://github.com/drupal/drupal/blob/10.1.x/core/lib/Drupal/Core/Config/ConfigFactoryInterface.php#L12-L21)

Reply:
You can try to run the following in a `drush php:cli` console:

```php
$config_manager = \Drupal::service('config.manager');
$config_factory = $config_manager->getConfigFactory();
$config = $config_factory->get('foo');
$config->getRawData();
```

The above will return a `\Drupal\Core\Config\ImmutableConfig` object even though `foo` is not a valid configuration entry. `$config->getRawData()` returns an empty array though.